From a8cab0bfe7a16d0ce32ad0d767b60cd9782fa409 Mon Sep 17 00:00:00 2001
From: Chen Liangjun <b36089@freescale.com>
Date: Tue, 7 Aug 2012 11:05:22 +0800
Subject: ENGR00219595 MX6Q ARM2: Add init data for P2P playback

Add init data for ESAI ASRC P2P playback in ARM2 board init file.

Signed-off-by: Chen Liangjun <b36089@freescale.com>
---
 arch/arm/mach-mx6/board-mx6q_arm2.c |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/mach-mx6/board-mx6q_arm2.c b/arch/arm/mach-mx6/board-mx6q_arm2.c
index 58edfad..5d9600d 100644
--- a/arch/arm/mach-mx6/board-mx6q_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6q_arm2.c
@@ -52,6 +52,8 @@
 #include <linux/regulator/fixed.h>
 #include <linux/mfd/max17135.h>
 #include <sound/pcm.h>
+#include <linux/mxc_asrc.h>
+
 
 #include <mach/common.h>
 #include <mach/hardware.h>
@@ -1557,8 +1559,14 @@ static const struct pm_platform_data mx6_arm2_pm_data __initconst = {
 	.suspend_exit	= arm2_suspend_exit,
 };
 
+static const struct asrc_p2p_params esai_p2p __initconst = {
+       .p2p_rate = 44100,
+       .p2p_width = ASRC_WIDTH_24_BIT,
+};
+
 static struct mxc_audio_platform_data sab_audio_data = {
 	.sysclk	= 16934400,
+	.priv = (void *)&esai_p2p,
 };
 
 static struct platform_device sab_audio_device = {
-- 
1.7.9.5

